### ISSUE TITLE
Run Dask Operator with read-only root filesystem

### ISSUE THREAD

**Issue by creste** [dask/dask-kubernetes#828]
Similar to #776 , many clusters prohibit running containers with writeable root filesystems. The Dask Operator's helm chart currently has `readOnlyRootFilesystem: true` commented out:
https://github.com/dask/dask-kubernetes/blob/c7839098e1c88f99d8110477981f9e7f3e6f49cc/dask_kubernetes/operator/deployment/helm/dask-kubernetes-operator/values.yaml#L31

I did a quick test with `readOnlyRootFilesystem: true` and the operator appears to work correctly.  Can the line above in `values.yaml` be uncommented?

**Comment by jacobtomlinson** [comment]
It looks like @samdyzon commented this out at some point, perhaps you can expand on why? I'm in favour of reinstating it if there are no negative impacts, it feels like a better security posture.

In the meantime @creste everything in `values.yaml` is user configurable. Is there anything stopping you from setting this to true when you install the chart?

**Comment by samdyzon** [comment]
Hi mate, not sure if there was any reason for disabling that setting - it think it might have been related to testing issues in CI but the commits were squashed so I can't be sure. If it doesn't break things you should definitely re-enable that setting for good security practices.

**Comment by creste** [comment]
@jacobtomlinson - Sure, I opened https://github.com/dask/dask-kubernetes/pull/830.

>Is there anything stopping you from setting this to true when you install the chart?

Nothing stopping me.  I have enabled `readOnlyRootFilesystem` in my `values.yaml`. It works fine. Example:
```yaml
securityContext:
  capabilities:
    drop:
      - ALL
  runAsNonRoot: true
  runAsUser: 1000
  runAsGroup: 1000
  allowPrivilegeEscalation: false
  readOnlyRootFilesystem: true
```
Note I also added `runAsGroup: 1000`, which also appears to work fine, but I didn't see group 1000 explicitly set in the [Dockerfile](https://github.com/dask/dask-kubernetes/blob/c7839098e1c88f99d8110477981f9e7f3e6f49cc/dask_kubernetes/operator/deployment/Dockerfile#L17) so I didn't mention it in the issue's description.  I'm happy to add `runAsGroup` and update the Dockerfile if you'd like.
